PhotoPlusBackend/photoplus/src/main/java/pl/polsl/photoplus/controllers/api/ImageController.java

```diff
@@ -46,10 +45,10 @@ public ImageController(final ImageService imageService, final PermissionEvaluato
     @PreAuthorize("@permissionEvaluatorService.hasPrivilege(authentication, 'image', 'post' )")
     public ResponseEntity postImage(@Image(service = ImageService.class) final MultipartFile file) throws IOException {
         final String entityCode = imageService.save(new ImageModelDto(null, file.getOriginalFilename(), file.getBytes()));
-        final HttpHeaders headers = new HttpHeaders();
+        final HashMap<String,String> response = new HashMap<>(1);
         final URI uri = linkTo(methodOn(this.getClass()).getSingle(entityCode)).toUri();
-        headers.add("Location", uri.toASCIIString());
-        return new ResponseEntity(headers, HttpStatus.CREATED);
+        response.put("self", uri.toASCIIString());
```

Comment:
due to RFC 2616 (current definiton of Hypertext Transfer Protocol standard) POST response should have empty body and Location header

Reply:
Fixed in a487057 (backend) and in 4813b1b (frontend).